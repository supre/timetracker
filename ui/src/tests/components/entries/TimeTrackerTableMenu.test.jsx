import React from "react";
import renderer from "react-test-renderer";
import { cleanup, fireEvent, render, screen } from "@testing-library/react";
import Menu from "./../../../components/entries/TimeTrackerTableMenu";

const onEvent = (e) => {
  console.log(e);
};

const Component = (props) => {
  const getVal = (key, d) => {
    return props && props[key] ? props[key] : d;
  };

  return (
    <Menu
      userId={1}
      onRefresh={onEvent}
      onDateChange={onEvent}
      onFilterClick={onEvent}
      isFiltered={getVal("isFiltered", false)}
      filterUrl={"http"}
      addEntryToList={onEvent}
      entries={[]}
      currentUser={{ id: 1 }}
    />
  );
};

test("Menu should render", () => {
  let menu = renderer.create(Component());
  let tree = menu.toJSON();
  expect(tree).toMatchSnapshot();
});

test("filtered is only enabled when flag is on", () => {
  render(Component());

  let container = screen.getByText("HTML");
  let parentLink = container.firstChild.parentElement;
  expect(parentLink.classList.contains("disabled")).toBeTruthy();

  cleanup();

  render(Component({ isFiltered: true }));

  container = screen.getByText("HTML");
  parentLink = container.firstChild.parentElement;

  expect(parentLink.classList.contains("disabled")).toBeFalsy();
});

test("Track time click should open entry model", () => {
  render(Component());

  let container = screen.getByText("Track Time");
  fireEvent.click(container);

  let modelContainer = screen.getByText("Track your time").parentElement
    .parentElement;

  expect(modelContainer).toMatchSnapshot();
});
