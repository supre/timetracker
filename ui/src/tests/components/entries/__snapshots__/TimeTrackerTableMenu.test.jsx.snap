// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu should render 1`] = `
<div
  className="ui compact inverted secondary top attached menu"
>
  <div
    className="right menu"
  >
    <div
      className="item"
      onClick={[Function]}
    >
      <button
        className="ui facebook circular icon inverted button"
        onClick={[Function]}
      >
        <i
          aria-hidden="true"
          className="refresh icon"
          onClick={[Function]}
        />
      </button>
    </div>
    <div
      className="item"
      onClick={[Function]}
    >
      <button
        className="ui facebook icon inverted right labeled button"
        onClick={[Function]}
      >
        <i
          aria-hidden="true"
          className="clock outline icon"
          onClick={[Function]}
        />
        Track Time
      </button>
    </div>
    <div
      className="item"
      onClick={[Function]}
    >
      <div
        className="field"
        style={
          Object {
            "display": "inline-block",
            "position": "relative",
          }
        }
      >
        <div
          className="field"
        >
          <div
            className="ui icon input"
            data-testid="datepicker-input"
          >
            <input
              disabled={false}
              onBlur={[Function]}
              onChange={[Function]}
              onClick={[Function]}
              onKeyDown={[Function]}
              placeholder="Select Range"
              readOnly={true}
              required={false}
              type="text"
              value=""
            />
            <i
              aria-hidden="true"
              className="calendar link icon"
              data-testid="datepicker-icon"
              onClick={[Function]}
            />
          </div>
        </div>
      </div>
      <div
        className="ui facebook icon right attached button"
        onClick={[Function]}
        role="button"
        tabIndex={0}
      >
        <i
          aria-hidden="true"
          className="filter icon"
          onClick={[Function]}
        />
      </div>
    </div>
    <div
      className="item"
      onClick={[Function]}
    >
      <a
        className="ui facebook icon disabled left labeled button"
        href="http"
        onClick={[Function]}
        role="button"
        tabIndex={-1}
        target="_BLANK"
      >
        <i
          aria-hidden="true"
          className="download icon"
          onClick={[Function]}
        />
        HTML
      </a>
    </div>
  </div>
</div>
`;

exports[`Track time click should open entry model 1`] = `
<div
  class="ui page modals dimmer transition visible active"
  style="display: flex;"
>
  <div
    class="ui basic modal transition visible active"
  >
    <div
      class="header"
    >
      <i
        aria-hidden="true"
        class="clock icon"
      />
      Track your time
    </div>
    <div
      class="content"
    >
      <form
        class="ui form entryModal"
      >
        <div
          class="field"
          style="display: inline-block; position: relative;"
        >
          <div
            class="field"
          >
            <div
              class="ui left icon input"
              data-testid="datepicker-input"
            >
              <input
                placeholder="Pick date"
                readonly=""
                type="text"
                value=""
              />
              <i
                aria-hidden="true"
                class="calendar link icon"
                data-testid="datepicker-icon"
              />
            </div>
          </div>
        </div>
        <div
          class="field"
        >
          <div
            class="ui fluid labeled input"
          >
            <div
              class="ui label label"
            >
              HoursWorked
            </div>
            <input
              name="hoursWorked"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class="field"
        >
          <textarea
            name="notes"
            placeholder="Write your notes..."
            rows="3"
          />
        </div>
      </form>
    </div>
    <div
      class="content"
    />
    <div
      class="actions"
    >
      <button
        class="ui green button"
      >
        <i
          aria-hidden="true"
          class="checkmark icon"
        />
         Add
      </button>
      <button
        class="ui red button"
      >
        <i
          aria-hidden="true"
          class="remove icon"
        />
         Cancel
      </button>
    </div>
  </div>
</div>
`;
